addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -65,6 +65,7 @@ def _holiday_status_id_domain(self):
     employee_id = fields.Many2one(
         'hr.employee', compute='_compute_from_holiday_type', store=True, string='Employee', index=True, readonly=False, ondelete="restrict", tracking=True,
         states={'cancel': [('readonly', True)], 'refuse': [('readonly', True)], 'validate1': [('readonly', True)], 'validate': [('readonly', True)]})
+    active = fields.Boolean(related='employee_id.active', readonly=True)
```

Comment:
I'm afraid that we mix the magic logic of the active field on both 2 models. The risk to bring unwanted side effect is probable:
- Here the field is readonly, but for instance, if we archive the allocation, we archive the employee, which is surely unwanted.
- Having a non stored active field could bring inconsistencies too.
- ...

In any case I find that it is clearer "active_employee" and it avoids side effects. So no reason not to do it ^^

To archive the allocation when archiving the employee, we could use the departure wizard don't you think ? 

![Michael-Lambert-Odoo](https://user-images.githubusercontent.com/8435879/119959091-f40d0300-bfa3-11eb-829b-83c9838e8305.png)